api/controllers/v1/jobs.ts

```diff
@@ -69,6 +71,10 @@ export const HandleJobs = async (event: any = {}): Promise<any> => {
           case JobStatus[JobStatus.completed]:
             queueUrl = c.get('jobUpdatesQueueUrl');
             await NotifyBuildSummary(jobId);
+            const environments = new Set(['dev', 'dotcomstg', 'dotcomprd']);
+            if (environments.has(c.get('env'))) {
+              await SubmitArchiveJob(jobId);
+            }
```

Comment:
I feel that this logic should exist in the `submitArchiveJob` function and not in this function - the batch service can hold the responsibility of determining whether or not the environment is valid for batch submission

Not a strongly held opinion, let me know if you disagree! 😄

Reply:
So i also think this logic is a little whacky, but i'd like to keep it as is. The problem is having environment conditional logic. But assuming that we only want the archive / rollbacks to happen for `dev` `dotcomstg` and `dotcomprd`, this place in the flow makes the most sense to me. I think it'd be worse to push it to the lower level functions, since it'd be confusing to see a function invocation `await SubmitArchiveJob(jobId)` and then potentially that function doesn't actually do anything.

Do you think it'd be better to remove the conditional logic all-together and create archive / deploy resources for `stg` and `prd`?

Reply:
I fully agree that ideally we don't have to manage a set of strings to match on 
I think the archive/deploy resources for environments would be ideal tbh

but I also think that `await SubmitArchiveJob(jobId)` wouldn't be confusing at all, since we don't assign a response or expect that function that mutate state within the function that's invoking it - as is the higher level function doesn't care or have responsibility over the results of the invocation

I'd rather have the environmental string matching in the lower level function, since if we truly want to only run on environment match, it makes sense to enforce it in the function executing the logic, not the invoker

But I'd rather not deal with the environment matching at all, so if it seems a reasonable lift, adding archive / deploy resources for `stg` and `prd` seems best

Reply:
sounds good. i'll create the resources and remove the conditional logic from this code.